packages/video_player_avplayer/tizen/src/media_player.cc

```diff
@@ -47,6 +48,34 @@ int64_t MediaPlayer::Create(const std::string &uri, int drm_type,
     return -1;
   }
 
+  if (!http_headers.empty()) {
+    auto iter = http_headers.find(flutter::EncodableValue("Cookie"));
+    if (iter != http_headers.end()) {
+      if (std::holds_alternative<std::string>(iter->second)) {
+        std::string cookie = std::get<std::string>(iter->second);
+        ret =
+            player_set_streaming_cookie(player_, cookie.c_str(), cookie.size());
+        if (ret != PLAYER_ERROR_NONE) {
+          LOG_ERROR("[MediaPlayer] player_set_streaming_cookie failed: %s.",
+                    get_error_message(ret));
+        }
+      }
+    }
+
+    iter = http_headers.find(flutter::EncodableValue("User-Agent"));
+    if (iter != http_headers.end()) {
+      if (std::holds_alternative<std::string>(iter->second)) {
+        std::string user_agent = std::get<std::string>(iter->second);
+        ret = player_set_streaming_user_agent(player_, user_agent.c_str(),
+                                              user_agent.size());
+        if (ret != PLAYER_ERROR_NONE) {
+          LOG_ERROR("[MediaPlayer] player_set_streaming_user_agent failed: %s.",
+                    get_error_message(ret));
+        }
+      }
+    }
+  }
+
```

Comment:
can we extract a new method for getting User-Agent and Cookie and move this method to VideoPlayer.